[강진호] Week4

jinhok96 opened the pull request:
## 요구사항

### 기본

- [x] 이메일 input에서 focus out 할 때, 값이 없을 경우 input에 빨강색 테두리와 아래에 “이메일을 입력해주세요.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일 input에서 focus out 할 때, 이메일 형식에 맞지 않는 값이 있는 경우 input에 빨강색 테두리와 아래에 “올바른 이메일 주소가 아닙니다.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일: test@codeit.com, 비밀번호: codeit101 으로 로그인 시도할 경우, “/folder” 페이지로 이동하나요?
- [x] 비밀번호 input에서 focus out 할 때, 값이 없을 경우 아래에 “비밀번호를 입력해주세요.” 에러 메세지가 보이나요?
- [x] 이외의 로그인 시도의 경우 이메일, 비밀번호 input에 빨강색 테두리와 각각의 input아래에 “이메일을 확인해주세요.”, “비밀번호를 확인해주세요.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일, 비밀번호 input에 에러 관련 디자인을 Components 영역의 에러 케이스로 적용했나요?
- [x] 로그인 버튼 클릭 또는 Enter키 입력으로 로그인 되나요?

### 심화

- [x] 눈 모양 아이콘 클릭시 비밀번호의 문자열이 보이기도 하고, 가려지기도 하나요?
- [x] 비밀번호의 문자열이 가려질 때는 눈 모양 아이콘에는 사선이 그어져있고, 비밀번호의 문자열이 보일 때는 사선이 없는 눈 모양 아이콘이 보이나요?

## 주요 변경사항

- /signin 페이지만 작업했습니다.

## 스크린샷

https://jinhok96-linkbrary.netlify.app/

## 멘토에게

- 제 계정으로 복사한 위클리 미션 Figma 파일 공유드립니다.
https://www.figma.com/file/thk8f0OClCyJasocUXtmEe/2%EC%A3%BC%EC%B0%A8-Weekly-Mission---Linkbrary?type=design&node-id=2%3A7940&mode=dev
- rebase를 했는데 커밋 충돌 수정하는 과정에서 커밋 내역이 지저분해진 것 같습니다. 적당히 걸러서 봐주세요...
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

jinhok96 committed:
new file:   css/common.css
	new file:   css/faq.css
	new file:   css/index.css
	new file:   css/privacy.css
	new file:   css/signin.css
	new file:   css/signup.css
	new file:   css/utility.css
	new file:   faq/faq.html
	new file:   index.html
	new file:   privacy/privacy.html
	new file:   signin/signin.html
	new file:   signup/signup.html
	new file:   src/akar-icons_facebook-fill.svg
	new file:   src/akar-icons_twitter-fill.svg
	new file:   src/akar-icons_youtube-fill.svg
	new file:   src/ant-design_instagram-filled.svg
	new file:   src/icon/Kakao.svg
	new file:   src/icon/download.svg
	new file:   src/icon/eye-off.svg
	new file:   src/icon/eye-on.svg
	new file:   src/icon/favorite.svg
	new file:   src/icon/google.svg
	new file:   src/icon/menu.svg
	new file:   src/landing-hero.png
	new file:   src/landing-section1.png
	new file:   src/landing-section2.png
	new file:   src/landing-section3.png
	new file:   src/landing-section4.png
	new file:   src/logo.svg
	new file:   src/metatag-twitter.png
	new file:   src/metatag.png

jinhok96 committed:
코드 정리
	modified:   css/common.css
	modified:   css/index.css
	modified:   css/signin.css
	modified:   css/signup.css
	modified:   css/utility.css

jinhok96 committed:
fix: signin, signup 모바일 크기에서 padding 줄어드는 문제 수정

jinhok96 committed:
refactor: 클래스명 직관적으로 변경 text-break -> mobile-only
	modified:   css/index.css
	modified:   index.html

jinhok96 committed:
refactor: font styling을 body로 이동, fallback 추가
	modified:   css/common.css

jinhok96 committed:
refactor: max-width를 pc크기로 이동
	modified:   css/common.css

jinhok96 committed:
refactor: feature-section 중앙 정렬 코드 수정
	modified:   css/common.css
	modified:   css/index.css

jinhok96 committed:
refactor: p태그 css reset 반영
	modified:   css/common.css

jinhok96 committed:
refactor: 분리된 gnb 관련 속성 하나로 결합
	modified:   css/common.css

jinhok96 committed:
refactor: common button -> .btn 변경
	modified:   css/common.css
	modified:   faq/faq.html
	modified:   index.html
	modified:   privacy/privacy.html
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: 링크 역할 button -> a태그로 변경
	modified:   faq/faq.html
	modified:   index.html
	modified:   privacy/privacy.html

jinhok96 committed:
fix: .btn 텍스트 데코 제거
	modified:   css/common.css

jinhok96 committed:
refactor: 필요없는 p태그 제거
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: button 태그로 .btn 속성 일부 이동
	modified:   css/common.css
	modified:   css/utility.css

jinhok96 committed:
refactor: .btn 확장성을 위해 속성 재분배
fix: a태그에 적용되는 button 속성을 .btn으로 롤백
	modified:   css/common.css
	modified:   css/utility.css

jinhok96 committed:
refactor: 버튼 확장성을 위해 일부 속성 이동
	modified:   css/common.css
	modified:   css/utility.css

jinhok96 committed:
refactor: 공통 적용 속성 하나로 통합
	modified:   css/utility.css

jinhok96 committed:
refactor: btn 색상 관련 클래스 수정
	modified:   css/common.css
	modified:   css/utility.css

jinhok96 committed:
refactor: 버튼 클래스 변경된 이름으로 수정

jinhok96 committed:
refactor: css reset으로 통합
	modified:   css/index.css

jinhok96 committed:
refactor: hero__title 클래스 추가
	modified:   index.html

jinhok96 committed:
refactor: feature__title 클래스 추가
	modified:   index.html

jinhok96 committed:
refactor: feature__description 클래스 추가
	modified:   index.html

jinhok96 committed:
fix: 누락된 클래스 추가
	modified:   index.html

jinhok96 committed:
refactor: h1 -> .hero__title 변경
	modified:   css/index.css

jinhok96 committed:
refactor: h2 -> .feature__title 변경
	modified:   css/index.css

jinhok96 committed:
refactor: h3 -> .feature__description 변경
	modified:   css/index.css

jinhok96 committed:
refactor: form-sign 클래스 추가
	modified:   signin/signin.html

jinhok96 committed:
refactor: form-sign 클래스 추가
	modified:   signup/signup.html

jinhok96 committed:
refactor: form -> .form-sign 변경
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor:  form-sign__label 클래스 추가
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: label -> .form-sign__label 변경
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
fix: 잘못된 transform 수정
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor: 필요없는 클래스 제거
	modified:   css/utility.css

jinhok96 committed:
refactor:  class="form-sign__input" 추가
	modified:   css/common.css
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: input -> .form-sign__input 변경
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
fix: 잘못된 focus 삭제 수정
	modified:   css/signup.css

jinhok96 committed:
refactor: 공통 속성 통합
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor: 필요없는 padding: 0 제거
	modified:   css/index.css
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor: css 각 html이 위치한 폴더로 이동
refactor: html, css로 연결되는 절대주소 -> 상대주소 변경
	renamed:    css/faq.css -> faq/faq.css
	modified:   faq/faq.html
	renamed:    css/index.css -> index.css
	modified:   index.html
	renamed:    css/privacy.css -> privacy/privacy.css
	modified:   privacy/privacy.html
	renamed:    css/signin.css -> signin/signin.css
	modified:   signin/signin.html
	renamed:    css/signup.css -> signup/signup.css
	modified:   signup/signup.html

jinhok96 committed:
refactor: src->public 폴더 이름 변경
refactor: src 절대 주소를 public 상대 주소로 수정
	modified:   faq/faq.html
	modified:   index.html
	modified:   privacy/privacy.html
	renamed:    src/akar-icons_facebook-fill.svg -> public/akar-icons_facebook-fill.svg
	renamed:    src/akar-icons_twitter-fill.svg -> public/akar-icons_twitter-fill.svg
	renamed:    src/akar-icons_youtube-fill.svg -> public/akar-icons_youtube-fill.svg
	renamed:    src/ant-design_instagram-filled.svg -> public/ant-design_instagram-filled.svg
	renamed:    src/icon/Kakao.svg -> public/icon/Kakao.svg
	renamed:    src/icon/download.svg -> public/icon/download.svg
	renamed:    src/icon/eye-off.svg -> public/icon/eye-off.svg
	renamed:    src/icon/eye-on.svg -> public/icon/eye-on.svg
	renamed:    src/icon/favorite.svg -> public/icon/favorite.svg
	renamed:    src/icon/google.svg -> public/icon/google.svg
	renamed:    src/icon/menu.svg -> public/icon/menu.svg
	renamed:    src/landing-hero.png -> public/landing-hero.png
	renamed:    src/landing-section1.png -> public/landing-section1.png
	renamed:    src/landing-section2.png -> public/landing-section2.png
	renamed:    src/landing-section3.png -> public/landing-section3.png
	renamed:    src/landing-section4.png -> public/landing-section4.png
	renamed:    src/logo.svg -> public/logo.svg
	renamed:    src/metatag-twitter.png -> public/metatag-twitter.png
	renamed:    src/metatag.png -> public/metatag.png
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: 파일 이름 일괄 변경
	renamed:    faq/faq.html -> faq/index.html
	renamed:    faq/faq.css -> faq/style.css
	modified:   index.html
	renamed:    privacy/privacy.html -> privacy/index.html
	renamed:    privacy/privacy.css -> privacy/style.css
	renamed:    signin/signin.html -> signin/index.html
	renamed:    signin/signin.css -> signin/style.css
	renamed:    signup/signup.html -> signup/index.html
	renamed:    signup/signup.css -> signup/style.css
	renamed:    index.css -> style.css

jinhok96 committed:
refactor: button->btn 변경
	modified:   signin/index.html
	modified:   signin/style.css
	modified:   signup/index.html
	modified:   signup/style.css

jinhok96 committed:
fix: 누락된 button->btn 변경 수정
	modified:   signin/style.css

jinhok96 committed:
feat: folder 페이지 추가

jinhok96 committed:
feat: .display-none 클래스 추가
refactor: 필요없는 폰트 속성 제거
	modified:   css/utility.css
	modified:   signin/style.css

jinhok96 committed:
feat: error-message 추가
refactor: input 속성 -> common/utility로 이동
	modified:   css/common.css
	modified:   css/utility.css
	modified:   signin/index.html
	modified:   signin/style.css

jinhok96 committed:
refactor: error-message 위치 이동
	modified:   css/utility.css
	modified:   signin/style.css

jinhok96 committed:
refactor: 필요없는 flex 관련 속성 제거
refactor: gap -> margin-top으로 변경
	modified:   signin/style.css

jinhok96 committed:
feat: display-none 클래스 적용
	modified:   signin/index.html

jinhok96 committed:
fix: btn-eye 위치 조정
	modified:   signin/style.css

jinhok96 committed:
feat: index.js 추가
	modified:   signin/index.html
	new file:   signin/index.js

jinhok96 committed:
refactor: btn-color-1 -> btn-color-primary 이름 변경
	modified:   css/utility.css
	modified:   faq/index.html
	modified:   folder/index.html
	modified:   privacy/index.html
	modified:   signin/index.html
	modified:   signin/index.js
	modified:   signup/index.html

jinhok96 committed:
new file:   css/_utility.css
	new file:   signin/_index.html
	new file:   signup/_index.html

jinhok96 committed:
new file:   css/common.css
	new file:   css/faq.css
	new file:   css/index.css
	new file:   css/privacy.css
	new file:   css/signin.css
	new file:   css/signup.css
	new file:   css/utility.css
	new file:   faq/faq.html
	new file:   index.html
	new file:   privacy/privacy.html
	new file:   signin/signin.html
	new file:   signup/signup.html
	new file:   src/akar-icons_facebook-fill.svg
	new file:   src/akar-icons_twitter-fill.svg
	new file:   src/akar-icons_youtube-fill.svg
	new file:   src/ant-design_instagram-filled.svg
	new file:   src/icon/Kakao.svg
	new file:   src/icon/download.svg
	new file:   src/icon/eye-off.svg
	new file:   src/icon/eye-on.svg
	new file:   src/icon/favorite.svg
	new file:   src/icon/google.svg
	new file:   src/icon/menu.svg
	new file:   src/landing-hero.png
	new file:   src/landing-section1.png
	new file:   src/landing-section2.png
	new file:   src/landing-section3.png
	new file:   src/landing-section4.png
	new file:   src/logo.svg
	new file:   src/metatag-twitter.png
	new file:   src/metatag.png

jinhok96 committed:
코드 정리
	modified:   css/common.css
	modified:   css/index.css
	modified:   css/signin.css
	modified:   css/signup.css
	modified:   css/utility.css

jinhok96 committed:
fix: signin, signup 모바일 크기에서 padding 줄어드는 문제 수정

jinhok96 committed:
refactor: 클래스명 직관적으로 변경 text-break -> mobile-only
	modified:   css/index.css
	modified:   index.html

jinhok96 committed:
refactor: feature-section 중앙 정렬 코드 수정
	modified:   css/common.css
	modified:   css/index.css

jinhok96 committed:
refactor: common button -> .btn 변경
	modified:   css/common.css
	modified:   faq/faq.html
	modified:   index.html
	modified:   privacy/privacy.html
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: 링크 역할 button -> a태그로 변경
	modified:   faq/faq.html
	modified:   index.html
	modified:   privacy/privacy.html

jinhok96 committed:
refactor: 필요없는 p태그 제거
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: button 태그로 .btn 속성 일부 이동
	modified:   css/common.css
	modified:   css/utility.css

jinhok96 committed:
refactor: form -> .form-sign 변경
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor:  class="form-sign__input" 추가
	modified:   css/common.css
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
refactor: h3 -> .feature__description 변경
	modified:   css/index.css

jinhok96 committed:
refactor: form-sign 클래스 추가
	modified:   signin/signin.html

jinhok96 committed:
refactor: h2 -> .feature__title 변경
	modified:   css/index.css

jinhok96 committed:
refactor: input -> .form-sign__input 변경
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor: 필요없는 padding: 0 제거
	modified:   css/index.css
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor: css reset으로 통합
	modified:   css/index.css

jinhok96 committed:
refactor: 필요없는 클래스 제거
	modified:   css/utility.css

jinhok96 committed:
refactor: 버튼 클래스 변경된 이름으로 수정

jinhok96 committed:
refactor: form-sign 클래스 추가
	modified:   signup/signup.html

jinhok96 committed:
refactor: label -> .form-sign__label 변경
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor: h1 -> .hero__title 변경
	modified:   css/index.css

jinhok96 committed:
fix: 잘못된 transform 수정
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor:  form-sign__label 클래스 추가
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
fix: 잘못된 focus 삭제 수정
	modified:   css/signup.css

jinhok96 committed:
refactor: 공통 속성 통합
	modified:   css/signin.css
	modified:   css/signup.css

jinhok96 committed:
refactor: css 각 html이 위치한 폴더로 이동
refactor: html, css로 연결되는 절대주소 -> 상대주소 변경
	renamed:    css/faq.css -> faq/faq.css
	modified:   faq/faq.html
	renamed:    css/index.css -> index.css
	modified:   index.html
	renamed:    css/privacy.css -> privacy/privacy.css
	modified:   privacy/privacy.html
	renamed:    css/signin.css -> signin/signin.css
	modified:   signin/signin.html
	renamed:    css/signup.css -> signup/signup.css
	modified:   signup/signup.html

jinhok96 committed:
refactor: src->public 폴더 이름 변경
refactor: src 절대 주소를 public 상대 주소로 수정
	modified:   faq/faq.html
	modified:   index.html
	modified:   privacy/privacy.html
	renamed:    src/akar-icons_facebook-fill.svg -> public/akar-icons_facebook-fill.svg
	renamed:    src/akar-icons_twitter-fill.svg -> public/akar-icons_twitter-fill.svg
	renamed:    src/akar-icons_youtube-fill.svg -> public/akar-icons_youtube-fill.svg
	renamed:    src/ant-design_instagram-filled.svg -> public/ant-design_instagram-filled.svg
	renamed:    src/icon/Kakao.svg -> public/icon/Kakao.svg
	renamed:    src/icon/download.svg -> public/icon/download.svg
	renamed:    src/icon/eye-off.svg -> public/icon/eye-off.svg
	renamed:    src/icon/eye-on.svg -> public/icon/eye-on.svg
	renamed:    src/icon/favorite.svg -> public/icon/favorite.svg
	renamed:    src/icon/google.svg -> public/icon/google.svg
	renamed:    src/icon/menu.svg -> public/icon/menu.svg
	renamed:    src/landing-hero.png -> public/landing-hero.png
	renamed:    src/landing-section1.png -> public/landing-section1.png
	renamed:    src/landing-section2.png -> public/landing-section2.png
	renamed:    src/landing-section3.png -> public/landing-section3.png
	renamed:    src/landing-section4.png -> public/landing-section4.png
	renamed:    src/logo.svg -> public/logo.svg
	renamed:    src/metatag-twitter.png -> public/metatag-twitter.png
	renamed:    src/metatag.png -> public/metatag.png
	modified:   signin/signin.html
	modified:   signup/signup.html

jinhok96 committed:
fix: 누락된 button->btn 변경 수정
	modified:   signin/style.css

jinhok96 committed:
refactor: button->btn 변경
	modified:   signin/index.html
	modified:   signin/style.css
	modified:   signup/index.html
	modified:   signup/style.css

jinhok96 committed:
new file:   css/_utility.css
	new file:   signin/_index.html
	new file:   signup/_index.html

jinhok96 committed:
refactor: 필요없는 flex 관련 속성 제거
refactor: gap -> margin-top으로 변경
	modified:   signin/style.css

jinhok96 committed:
refactor: btn-color-1 -> btn-color-primary 이름 변경
	modified:   css/utility.css
	modified:   faq/index.html
	modified:   folder/index.html
	modified:   privacy/index.html
	modified:   signin/index.html
	modified:   signin/index.js
	modified:   signup/index.html

jinhok96 committed:
refactor: 파일 이름 일괄 변경
	renamed:    faq/faq.html -> faq/index.html
	renamed:    faq/faq.css -> faq/style.css
	modified:   index.html
	renamed:    privacy/privacy.html -> privacy/index.html
	renamed:    privacy/privacy.css -> privacy/style.css
	renamed:    signin/signin.html -> signin/index.html
	renamed:    signin/signin.css -> signin/style.css
	renamed:    signup/signup.html -> signup/index.html
	renamed:    signup/signup.css -> signup/style.css
	renamed:    index.css -> style.css

jinhok96 committed:
feat: folder 페이지 추가

jinhok96 committed:
feat: .display-none 클래스 추가
refactor: 필요없는 폰트 속성 제거
	modified:   css/utility.css
	modified:   signin/style.css

jinhok96 committed:
feat: index.js 추가
	modified:   signin/index.html
	new file:   signin/index.js

jinhok96 committed:
fix: btn-eye 위치 조정
	modified:   signin/style.css

jinhok96 committed:
refactor: error-message 위치 이동
	modified:   css/utility.css
	modified:   signin/style.css

jinhok96 committed:
feat: display-none 클래스 적용
	modified:   signin/index.html

jinhok96 committed:
feat: error-message 추가
refactor: input 속성 -> common/utility로 이동
	modified:   css/common.css
	modified:   css/utility.css
	modified:   signin/index.html
	modified:   signin/style.css

jinhok96 committed:
Merge branch 'part1-강진호-week4' of https://github.com/jinhok96/4-Weekly-Mission-jinhok96 into part1-강진호-week4

jinhok96 committed:
fix: btn-color-1 -> btn-color-primary 누락된 수정
	modified:   index.html

jinhok96 committed:
fix: 백업 파일 복구
	deleted:    css/_utility.css
	modified:   css/utility.css
	deleted:    signin/_index.html
	modified:   signin/index.html

jinhok96 committed:
Revert "fix: btn-color-1 -> btn-color-primary 누락된 수정"

This reverts commit 6e51f9ac0290e0db02caa31e2f50fd17104702a6.

jinhok96 committed:
fix: rebase 중 누락된 부분 수정
	modified:   css/common.css
	modified:   css/utility.css
	modified:   index.html
	modified:   signin/index.html
	deleted:    signup/_index.html

jinhok96 committed:
feat: signin페이지 유효성 검사 스크립트 추가
	modified:   signin/index.js

jinhok96 committed:
feat: 로그인 성공 시 folder로 연결
	modified:   folder/index.html
	modified:   signin/index.js

jinhok96 committed:
feat: email, password에 id 추가
refactor: js email, password querySelector 변경
	modified:   signin/index.html
	modified:   signin/index.js

jinhok96 committed:
feat: password btn-eye 토글 기능 추가
	modified:   signin/index.html
	modified:   signin/index.js

jinhok96 committed:
chore: 주석 정리
	modified:   signin/index.js

jinhok96 committed:
feat: 에러 메시지 출력 시 인풋 테두리 red색상
	modified:   css/common.css
	modified:   signin/index.js

jinhok96 committed:
feat: 사용자 입력 제어 함수 추가 - click, Enter
feat: 로그인버튼 이벤트리스너 handleUserAction으로 통일
	modified:   signin/index.js

1005hoon commented on css/common.css:
```diff
@@ -117,6 +117,30 @@ button {
+.input-error {
+  outline: 0.1rem solid var(--red, #ff5b56);
+}
+
```
input 요소는 다양한 곳에서 활용될수 있습니다! 
예를 들어, 로그인을 위한 폼 인풋으로도 활용될 수 있고, 
만약 채팅입력을 한다면 채팅 입력창도 인풋이 될 수 있죠. 

그런만큼, Input element가 기본적으로 가져야 할 스타일에 대해서는 완전 skeleton 또는 primitive 급으로 가져야 하는게 좋을것 같은데요. 여기서 스켈레톤 또는 프리미티브라 함은, 이 요소가 완전 원시적으로, 그리고 기본적으로 지녀야만 하는, 모든 곳에서 똑같이 활용되는 스타일만 가져야 한다는 뜻입니다. 

지금은 signin 폼에서만 활용이 되었기 때문에 이 공통 스타일링을 활용할 수 있겠지만, 이후 프로젝트가 고도화 됨에 따라 검색창을 구현해야 하거나 하는 경우에도 이와 같은 스타일링을 가질지는 의문입니다. 

따라서 공통 Input 요소는 기본적인 넓이와 패딩, Radius와 같은 값들과 기초 색상을 설정하고, Color와 border color, 백그라운드 색등은 common color style을 만들어 활용하는게 조금 더 안전해 보입니다. 

예를 들자면 

로그인 폼의 인풋은 기본적인 Input의 스타일에서 특정 색과 Focus 상태가 필요하다면 이에 관해서는 `login-form__input` 이라는 클래스를 signin/style.css파일에 만들고 다음과 같이 사용할 수 있겠죠. 

```
.login-form__input { ... }
.login-form__input:hover,
.login-form__input:focus { ... }

<input class="login-form__input text-black bg-white border-primary" />

const emailInput = document.querySelector('.login-form__input'); 
... 이메일 체크 로직 ... 
emailInput.addClass... = "border-error"
```

1005hoon commented on css/utility.css:
```diff
@@ -72,3 +72,7 @@
     width: 35rem;
   }
 }
+
```
이 btn-color-primary 가 활용되는곳이 많은가요? 
만약 모든곳에서 공통된다면 그냥 `btn-primary` 라는 콤포넌트성 클래스 이름으로 변경해도 괜찮겠다 싶습니다. 
그게 아니라면, 기본적으로 버튼이 지녀야 할 스타일을 담은 유틸 클래스 `btn` 과 
`text-primary` 로 버튼 내부 텍스트의 색을 조합해주고 
`bg-gradient-???` 라는 클래스로 버튼으 ㅣ색상을 조합해주어 활용하는 형태가 조금 더 확장성 높은 스타일 작성법일것 같아요. 
예를 들면 이런식으로 활용이 가능하겠죠 

```
.btn { ... 버튼 기본 스타일링 }
.text-white { color: #fff };
.bg-gradient-primary { background: var(--gra-purpleblue-to-skyblue, linear-gradient(91deg, #6d6afe 0.12%, #6ae3fe 101.84%)) }

<button class="btn text-white bg-gradient-primary">로그인</button>
```

1005hoon commented on css/utility.css:
```diff
@@ -72,3 +72,7 @@
+
+.display-none {
+  display: none;
+}
```
좋습니다! 
음 조금 더 직관적으로 생각해보면, 화면에서 숨기기위한 유틸성 클래스니까 `hidden` 이라는 클래스 이름은 어떤가요? ㅋㅋㅋ 한번 고려해봐도 좋을것 같습니다!

1005hoon replied:
`display: none` 이면 `hidden` 
`display: block` 이면 `block` 
`display: flex`면 `flex` !

1005hoon commented on faq/index.html:
```diff
@@ -29,7 +29,7 @@
-        <a class="btn btn-login btn-color-1" href="../signin/index.html">
+        <a class="btn btn-login btn-color-primary" href="../signin/index.html">
           로그인
         </a>
```
굳굳 이렇게 ui성 클래스 btn과 로그인 버튼이 지녀야할 `btn-login` 스타일, 그리고 색과 백그라운드 조합 형태로 하면 완벽할것 같아요

1005hoon commented on signin/index.html:
```diff
@@ -35,21 +35,23 @@
         </div>
+        <p id="error-password" class="error-message display-none">에러 메시지</p>
       </label>
     </form>
```
로그인 폼의 경우, 어차피 이게 폼인건 누구나 알 수 있습니다! 그래서 Form-sign이라는 이름은 조금 반복적으로 보여요. 
그리고 Label의 경우 Inline element인만큼 label 안에 div와 다양한 요소가 들어가는건 별로 좋아보이지 않습니다. 
만약 라벨 클릭 시 input에 포커스를 주고싶은 목적이였다면, label의 For 어트리뷰트를 활용해보세요 !

이런식으로 마크업을 해보는건 어떠신가요? 

```
<form class="signin flex flex-col gap-2">
  <div class="form-group flex flex-col gap-1"> -- 이메일에 필요한 요소를 그룹화 하기 위한 form group목적의 div
    <label class="text-sm text-light" for="email"> -- 라벨의 텍스트 크기와 색상 설정, 그리고 for 프로퍼티 추가 필요해요
      이메일
    </label>
    <input class="signin-input" name="email" type="email" />
    <p class="hidden text-error text-xs">오류 메시지 </ p>
  </div>
</ form>
```

1005hoon commented on signin/index.html:
```diff
@@ -35,21 +35,23 @@
 
-    <div class="button-wrap">
-      <button class="btn btn-login btn-color-1" type="button">로그인</button>
+    <div class="btn-wrap">
```
이녀석도 btn-wrap보다는 social-login이나 social-login-box 같은 이름이 조금 더 좋을것같아요

1005hoon replied:
음 지금보니까 이메일 로그인 버튼이 Form 밖으로 꺼내져있네요! 

이 버튼은 Form 내부에 존재해야 할것 같습니다. 
Form 안에 버튼이 있는 경우, 버튼은 기본적으로 `type="submit"` 을 지니고 있어서 클릭과 함께 폼 제출이 되거든요. 그리고 그게 조금 더 시맨틱하게 느껴집니다. 

따라서 로그인 버튼은 `signin` 폼 내부에 포함이 되어야 하고, 

소셜 로그인은 지금처럼 외부로 꺼내두어 자체만의 콤포넌트로 구현이 되는게 맞아보여요!

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+/********************
+ * VARIABLE DEFINTION
+ ********************/
+
```
ㅋㅋㅋㅋ 굳굳

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+    element.classList.add("display-none");
+    prevInput.classList.remove("input-error");
+  }
+}
```
오 함수명 좋습니다. 그리고 element와 message를 인자로 받아오는데요, 조금만 더 구분을 해보면 좋을것 같아요! 

지금 이 함수의 역할과 책임을 봐보면
1. 클래스명을 토글시킬 html요소를 입력받는다
2. html요소가 있고, 이게 emailErrorMessage 객체라면 email인풋을 선택하고, 아니라면 Password인풋을 선택한다??? < 너무 번잡합니다!
3. 전달받은 메시지가 있다면 ?????? 오잉  그리고 없다면???? 


조금 직관적이지가 않죠.

개선해본다면 다음과 같이 역할을 구분할 수 있겠네요 

```
function toggleInputError(errorElement, inputElement) {
  if (!errorElement) {
     // 전달받은 요소가 제대로된 html 태그가 아니라면 개발하는 단계에서 캐치할 수 있도록 오류를 던저줍니다
    throw new Error("error element가 존재하지 않습니다")
  }

  if (!inputElement) {
     // 전달받은 요소가 제대로된 html 태그가 아니라면 개발하는 단계에서 캐치할 수 있도록 오류를 던저줍니다
    throw new Error("input element가 존재하지 않습니다")
  }

  const isHidden = errorElement.classList.contains("hidden")
  if(isHidden) { 에러가 보이지 않는다면, 에러 보여주고 }
  else { 에러가 보이는 상태라면 에러 숨기고! }
}
```

이처럼 에러 토글하는 함수는, 어떤 element를 사용해야하는지 판단하지 않고, 그냥 전달받은 element가 제대로 된건지만 체크하고 class를 토글링 하는 작업만 처리하도록 하는게 좋아보입니다

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+    toggleErrorMessage(errorMessage);
+  }
+}
+
```
여기도 마찬가지에요! checkForm이라는 이름보다는 다양한 밸리데이션 함수를 만들고, 그에따라 호출되는 식으로 하는게 좋겠네요. 

```
function isInputEmpty(inputElement) {
  if (!inputElement) { 인풋 엘리먼트가 없습니다 오류 던지기 }
  const inputContent = inputElement.value; 
  if (inputContent.length === 0) {
    // 비어있다면 return false
    return false
  }
  // 아니면 Return true
  return true
}

function isEmailValid() { 
 이메일 validation 로직 구현
 만약 Valid하면 return true 
 else return false 
}
function isValidPassword() 이것도 마찬가지...

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+    toggleErrorMessage(passwordErrorMessage, "비밀번호를 확인해 주세요.");
+  }
+}
+
```
이 함수의 경우, 로그인 유효성 검사뿐만 아니라 로그인이 되면 페이지 이동까지 하고 있죠! 
그래서 조금 구분해도 좋을것 같습니다. 

```
function onLogin(email, password) {
  const isEmailEmpty = isInputEmpty(email) 
  const isValidEmailFormat = isEmailFormatted(email)
  const emailExists = isValidEmail(email) // 여기서 이메일이 test@codeit.com`인지 체크

  비밀번호도 마찬가지
  그리구 그에 따라서 에러메시지 토글링!
}
```

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+    icon.src = "../public/icon/eye-off.svg";
+  }
+}
+
```
showPassword와 HidePassword 함수로 분리하는게 좋아보여요

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+    element.preventDefault();
+  }
+}
+
```
handleUserAction보다는 onInputFocus와 onUserSubmit 과 같은 함수로 분리해서 구현하는게 좋아보입니다

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+password.addEventListener("focusout", function () {
+  checkForm(password);
+});
+
```
굳굳!

1005hoon reviewed (changes requested):
전반적으로 너무 잘 작성하셨습니다 ! 
많이 고민하시고 작성한게 보이는 코드였고, 조금 더 개선시킬 수 있는 방향을 정리해보자면 다음과 같습니다.

1. 마크업 개편
2. 공통 스타일링
3. 조금 더 Modular된 자바스크립트 구현

이에 대한 상세한 내용은 코드단위에서 설명이 되어있으니 참고 부탁드리구요, 
혹시 이해 안되는 부분이 있으시다면 디스코드 채널에 바로 질문주셔요!

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,115 @@
+//click, keypress 이벤트 핸들러
+loginBtn.addEventListener("click", handleUserAction);
+document.addEventListener("keypress", handleUserAction);
+eyeBtn.addEventListener("click", togglePassword);
```
만약 조금 더 나가서 제가 구조를 짠다면 이런식으로 짜볼것 같아요!  @jinhok96 

```js
/**
 * HTML DOM Elements
 */
const emailInput = document.querySelector('#email-input')
const passwordInput = document.querySelector('#password-input')
const loginButton = document.querySelector('#login-button')
const showPasswordButton = document.querySelector(".eyeoff");
/**
 * UI Event Listeners
 */

/**
 * Utility Logics
 */
function isEmailFormat(emailString) {
  const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
  return regex.test(emailString)
}

function isEmpty(text) {
  return text.length === 0
}

/**
 * Validates a user's login information.
 *
 * This function checks if the provided login information matches a predefined valid user schema.
 * It compares both email and password from the loginInfo object with the valid user credentials.
 *
 * @param {Object} loginInfo - An object containing the user's login information.
 * @param {string} loginInfo.email - The email address of the user.
 * @param {string} loginInfo.password - The password of the user.
 * @returns {boolean} - Returns `true` if the email and password match the valid user schema, otherwise returns `false`.
 */
function isValidUser(loginInfo) {
  const user = { email: "codeit@codeit.com", password: "password" }  // 나중에 데이터베이스 붙으면 이걸 네트워크 통신해서 받아오는 사용자 정보로 활용하면 되겠죠?

  const { email, password } = loginInfo
  
  if (email !== validUserSchema.email) {
    return false
  }

  if (password !== validUserSchema.password) {
    return false
  }

  return true
}

function validateUserLoginInfo() {
  const email = emailInput.value; 

  const isEmailEmpty = isEmpty(email)
  if (isEmailEmpty) {
    toggleError(emailErrorMessage, "이메일을 입력하세요")
    return false
  }

  const isValidEmailFormat = isEmailFormat(email)
  if (!isValidEmailFormat) {
    toggleError(emailErrorMessage, '잘못된 이메일 유형입니다')
    return false
  }

  const password = passwordInput.value;
  const isPasswordEmpty = isEmpty(password)
  if (isPasswordEmpty) {
    toggleError(passwordErrorMessage, "비밀번호를 입력하세요")
    return false
  }

  return true
}
/**
 * Service Logic
 */
function handleLogin(e) {
  const isValidInput = validateUserLoginInfo()
  if (!isValidInput) {
    return false
  }

  // login 로직 실행
  const user = isValidUser({ email, password });
  
  if (!user) {
    toggleError(emailErrorMessage, "이메일을 확인하세요")
    toggleError(passwordErrorMessage, "비밀번호를 확인하세요")
    return
  }

  location.href = "로그인 성공 후 이동 될 엔드포인트"
}
```

jinhok96 committed:
chore: btn-wrap -> social-login 명시적으로 변경
	modified:   signin/index.html
	modified:   signin/style.css
	modified:   signup/index.html
	modified:   signup/style.css

jinhok96 committed:
chore: 로그인 버튼 위치 form 내부로 이동
	modified:   signin/index.html
	modified:   signup/index.html

jinhok96 committed:
chore: social -> social-login에 통합
	modified:   signin/index.html
	modified:   signin/style.css
	modified:   signup/index.html
	modified:   signup/style.css

jinhok96 committed:
chore: form-sign -> sign으로 변경
	modified:   signin/index.html
	modified:   signin/style.css
	modified:   signup/index.html
	modified:   signup/style.css

jinhok96 committed:
chore: id, class 이름 수정
refactor: label과 관련된 구조 수정
chore: 위 변화에 따른 css 수정
	modified:   css/utility.css
	modified:   signin/index.html
	modified:   signin/style.css
	modified:   signup/index.html
	modified:   signup/style.css

jinhok96 committed:
chore: display-none -> hidden 이름 변경
	modified:   css/utility.css
	modified:   signin/index.html
	modified:   signin/index.js
	modified:   signin/style.css
	modified:   signup/index.html
	modified:   signup/style.css

jinhok96 committed:
refactor: common에 맞지 않는 클래스 이동
	modified:   css/common.css
	modified:   signin/index.js
	modified:   signin/style.css
	modified:   signup/style.css

jinhok96 committed:
feat: 여러 utility 클래스 추가
	modified:   css/utility.css

jinhok96 committed:
chore: 일부 필요없는 커밋 롤백
chore: 버튼색과 글자색 분리
	modified:   css/utility.css
	modified:   index.html
	modified:   signin/index.html
	modified:   signin/style.css
	modified:   signup/index.html

jinhok96 committed:
fix: 누락된 text-grey-light 추가
	modified:   faq/index.html
	modified:   folder/index.html
	modified:   privacy/index.html

jinhok96 committed:
refactor: login 페이지 javascript 재작성
- 함수 기능별로 세분화
- 코드 깔끔하게 정리
	modified:   signin/index.js

1005hoon commented on css/common.css:
```diff
@@ -117,6 +117,22 @@ button {
+input::placeholder {
+  color: var(--Linkbrary-gray60, #9fa6b2);
+}
+
```
굳!

1005hoon commented on css/utility.css:
```diff
@@ -47,8 +47,11 @@
+.text-grey-light {
   color: var(--Grey-Light, #f5f5f5);
+}
+
```
굳굳!!

1005hoon commented on css/utility.css:
```diff
@@ -47,8 +47,11 @@
+
+.btn-color-primary {
   background: var(--gra-purpleblue-to-skyblue, linear-gradient(91deg, #6d6afe 0.12%, #6ae3fe 101.84%));
 }
```
얘의 경우, `btn-color-primary` 보단 백그라운드에 그라디엔트 색이니 `bg-gradient--primary` 같은 이름으로 두고, 만약 다른곳에서 활용되는 경우, 다른곳에서도 쓸 수 있도록 해보면 어떨가요? 

만약 특정 버튼에서만 사용된다면 `btn-bg-primary` 가 조금 더 우리가 쓰고있는 형태와 맞을것 같아요!

1005hoon commented on css/utility.css:
```diff
@@ -72,3 +75,15 @@
+
+.p-relative {
+  position: relative;
+}
```
`relative` 라는 값은 `position` 속성에만 존재하니 `p-relative` 를 그냥 `relative` 라고만 해도 괜찮을것 같아요

1005hoon commented on faq/index.html:
```diff
@@ -29,7 +29,7 @@
       <gnb>
```
ㅋㅋ gnb라는 태그는 아쉽게도 없습니다 ㅜㅜ nav로 변경해보면 어떨까요?

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,163 @@
+
+//비밀번호 btn-eye
+const eyeBtn = document.querySelector(".btn-eye");
+
```
UI 요소를 선택한 변수의 경우, 뒤에 `element` 를 추가해주어 실제 에러메시지 등과 구분하기 쉽게 해주면 좋을것 같아요 
`emailErrorMessage` 라는 이름을 나중에 누군가가 본다면 이 변수에 저장된 값이 p태그인지, 아니면 메시지 콘텐츠 그 자체인지 파악하기가 어렵잖아요 ㅎㅎ

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,163 @@
+  email: "test@codeit.com",
+  password: "codeit101",
+};
+
```
상수로 꺼내둔거 너무 좋고! 다만 이렇게 상수를 활용하는 경우, 다 대문자와 `_` 를 사용합니다. 
이 경우, 데이터베이스에 저장된 사용자 정보를 사용하는 대신, 임의로 이 유저 정보를 사용하고 있죠? 
그래서 `const VALID_USER = { ... }` 로 두고, 
```
function getUserByLoginInfo({ email, password }) {
  if(email === VALID_USER.email && password === VALID_USER.password) {
    return VALID_USER
  }

  return null
}
```

로 두어서 활용하면 좋을것 같습니다.

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,163 @@
+  }
+  return false;
+}
+
```
인풋 요소를 전달받기보단 값만 전달받아서 그 값에 대한 validation을 돌리는게 조금 더 좋아보입니다

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,163 @@
+}
+
+//이메일 양식
+function isEmailFormat(emailString) {
```
요 함수로 분리하는거 좋습니다! 이 때, `validateEmail` 과 같은 함수명이 조금 더 적합해 보여요!

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,163 @@
+  } else {
+    return null;
+  }
+}
```
요고는 이렇게 모든걸 처리해주는 마법같은 함수로 두기보단, 개발자가 특정 케이스에 해당하는 인풋과 태그들을 직접 선택해 할당하도록 하는게 더 명시적이고 좋아보여요

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,163 @@
+
+//eyeBtn 비밀번호 보이게
+function showPassword() {
+  const icon = eyeBtn.firstElementChild;
```
first elemenChild 로 선택하는건 살짝 위험할수도 있을것 같아요! 
오히려 `const imgElement = document.querySelector(".eye-btn img")` 같은 형태로 선택자를 쓰면 어떤가요?

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,163 @@
+  } else {
+    hideError(emailErrorMessage);
+  }
+}
```
if else의 조합도 좋지만, 조금 더 욕심내어본다면 [early return 식을 활용하면 어떨까요](https://dev.to/shameel/-clean-coding-tip-early-return-principle-47kj)? 

```
  if (isEmpty(emailInput)) {
    return showError(emailErrorMessage, "이메일을 입력해 주세요.");
  } 
  if (!isValidEmail()) {
    return showError(emailErrorMessage, "올바른 이메일 주소가 아닙니다.");
  } 

  return hideError(emailErrorMessage);
}
```

1005hoon reviewed (changes requested):
너무 좋습니다! 깔끔하고 코드퀄리티도 많이 올라왔네요.
조금 더 모듈럴 하게 코드를 작성하고 가독성을 키우기 위해 early return 형태를 사용해본다면 훨씬 더 관리하기 좋은 코드로 업그레이드가 될것 같아요! 

저번 멘토링 세션때 이야기 드렸던 요소들을 잘 팔로우 하신게 느껴지는 코드네요 ㅎㅎ 
코드단에서 리뷰 남겨두었으니 참고해서 작업해주시고, 궁금한거 있으시면 디스코드에 문의주셔요

jinhok96 committed:
fix: 잘못된 gnb 태그 수정
	modified:   faq/index.html
	modified:   folder/index.html
	modified:   privacy/index.html

jinhok96 committed:
chore: errorMessage -> errorMessageElement 변경
	modified:   signin/index.js

jinhok96 committed:
chore: 상수 loginInfo -> VALID_USER로 변경
	modified:   signin/index.js

jinhok96 committed:
chore: 유저 로그인 알고리즘 개선
	modified:   signin/index.js

jinhok96 committed:
chore: input 파라미터를 input.value로 변경
	modified:   signin/index.js

jinhok96 committed:
chore: isEmailFormat -> validateEmail 이름 변경
	modified:   signin/index.js

jinhok96 committed:
chore: firstElementChild를 안전한 방식으로 변경
	modified:   signin/index.js

jinhok96 committed:
chore: ACTIVE FUNCTION에 대해 early return 적용
fix: imgElement에 대한 잘못된 querySelector 수정
	modified:   signin/index.js

jinhok96 committed:
chore: targetInput 함수 제거
chore: showError, hideError에 input 파라미터 추가
	modified:   signin/index.js

1005hoon commented on signin/index.js:
```diff
@@ -0,0 +1,155 @@
+    return true;
+  }
+  return false;
+}
```
문득 생각해보니 이것도 `emailInput` DOM element를 validateEmail에 바로 곶아주기보단, `isValidEmail` 에 이메일 값을 전달받아 제공하는 형태가 되어주는게 좋을것 같아요

1005hoon reviewed (approved):
오 너무 좋은데요 ㅋㅋ
4주차 작업은 여기서 마무리짓도록 할게요
고생하셨습니다 👍🏻
